File: Python/Exor.py
testCase = int(input())
li = []
for i in range(0,testCase):
    n = int(input())
    # a = [int(x) for x in input().split(None)]
    a = input().split(None)
    a = list(map(lambda x : int(x) , a ))
    a.insert(0,0)
    a.append(0)
    a.append(0)
    count = 0
    count1 =0
    count2 = 0
    j = 1
    while j != n+1:
        k = j+1
        while k != n+1:
            l = k
            while l != n+1 :
                try:
                    for z in range(j,k):
                        count1 = count1^a[z]
                    for z in range(k,l+1):
                        count2 = count2^a[z]
                    if count1 == count2:
                        count = count +1
                    count1 = 0
                    count2 = 0
                except Exception as e:
                    pass
                l = l +1
            k = k+ 1
        j = j+ 1
    del a
    a = None
    li.append(count)
for x in li:
    print(x)

File: Java/ExOr.java
import java.util.*;
import java.io.*;
public class ExOr{
  public static void main(String[] args) throws IOException {
    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    int testCase = Integer.parseInt(br.readLine());

    StringBuilder sb = new StringBuilder();
    for (int a =0;a< testCase ;a++ ) {
      int count = 0;
      int n = Integer.parseInt(br.readLine());
      int[] z = new int[n];
      String s = br.readLine();
      String[] token = s.split(" ");
      for(int j = 0 ;j<n ; j++){
        z[j] = Integer.parseInt(token[j]);
      }
      for(int i = 0 ; i<n ;i ++){
        for(int j = 0 ;j<n ; j++){
          for(int k = 0 ; k<n ;k++){
            // if( (i+1>=n)|| (j-1)<=0 || (j+1) >= n ){
            //   continue;
            // }
            // else{
            try {
              if( (z[i]^z[i+1]^z[j-1]) == (z[j]^z[j+1]^z[k]) ){
                count ++;
              // }
            }
          }
            catch(Exception e) {}
          }
        }
      }
      sb.append(count);
      sb.append("\n");
    }
    System.out.println(sb.toString());
  }
}
